lib/pages/timetable/view/timetable_page.dart

```diff
@@ -291,6 +291,30 @@ class _TimetablePageState extends State<TimetablePage> {
           )
         ],
       );
+    } else if (!user.sources.contains('students')) {
+      return AppDialog(
+        title: S.current.warningNoEvents,
+        content: [
+          Text(
+            S.current.warningNoStudentSource,
+          ),
+        ],
+        actions: [
+          AppButton(
+            text: S.current.settingsSourceSelection,
+            width: 130,
+            onTap: () async {
+              Navigator.of(context).pop();
+              if (authProvider.isAnonymous) {
```

Comment:
This dialog shouldn't show up if the user is anonymous, right?

Reply:
![image](https://user-images.githubusercontent.com/24855378/167191281-b0829160-da0d-4079-824f-67718ff9fa7c.png)

as far as I understand, an anounymous user should not reach the timetable section

I cannot fully understand the source code for this section to be honest, so I am not able to give an accurate review. I do not understand how specific app-dialogs should be triggered

Reply:
maybe @AdrianMargineanu  could look into this

Reply:
Solved